src/wallet/test/wallet_tests.cpp

```diff
@@ -31,6 +34,38 @@ extern UniValue getnewaddress(const JSONRPCRequest& request);
 
 BOOST_FIXTURE_TEST_SUITE(wallet_tests, WalletTestingSetup)
 
+static std::shared_ptr<CWallet> TestLoadWallet(interfaces::Chain& chain)
+{
+    bilingual_str error;
+    std::vector<bilingual_str> warnings;
+    auto wallet = CWallet::CreateWalletFromFile(chain, WalletLocation(""), error, warnings);
+    wallet->postInitProcess();
+    return wallet;
+}
+
+static void TestUnloadWallet(std::shared_ptr<CWallet>&& wallet)
+{
+    std::vector<bilingual_str> warnings;
+    SyncWithValidationInterfaceQueue();
+    wallet->m_chain_notifications_handler.reset();
+    RemoveWallet(wallet, nullopt, warnings);
```

Comment:
anyone remembers, why it requires here `RemoveWallet` ?

https://github.com/bitcoin/bitcoin/pull/18727/files#diff-157f588a09ff1da05b8c74acfcfb4da21917b6a97ed5d2e702228d621d23e66dR40-R44